pkg/iris-middleware/oapi_validate.go

```diff
@@ -0,0 +1,184 @@
+package middleware
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/getkin/kin-openapi/openapi3filter"
+	"github.com/getkin/kin-openapi/routers"
+	"github.com/getkin/kin-openapi/routers/gorillamux"
+	"github.com/kataras/iris/v12"
+)
+
+const (
+	IrisContextKey = "oapi-codegen/iris-context"
+	UserDataKey    = "oapi-codegen/user-data"
+)
+
+// OapiValidatorFromYamlFile creates a validator middleware from a YAML file path
+func OapiValidatorFromYamlFile(path string) (iris.Handler, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("error reading %s: %s", path, err)
```

Comment:
Could we use native error wrapping?

```suggestion
		return nil, fmt.Errorf("error reading %s: %w", path, err)
```